### ISSUE TITLE
Cannot reproduce the results from the paper - Problem with InPlaceABNSync?

### ISSUE THREAD

**Issue by alexandrewillame** [PeikeLi/Self-Correction-Human-Parsing#88]
I cannot reproduce the results from the paper. I suspect it is a problem with the InPlaceABNSync function.

The code as-is does not work on my computer (either on Windows 11 or WSL = Linux), some problem with Ninja / C++ compiling. I want to get rid of the code that uses Ninja which is the code of the _InPlaceABNSync_ function.

I thus modified the AugmentCE2P to bypass InPlaceABNSync which is the module that uses Ninja/cpp functions.. I modified `networks/AugmentCE2P.py`, line 21 and 23:

```
from modules import InPlaceABNSync

BatchNorm2d = functools.partial(InPlaceABNSync, activation='none')
```

I deleted these lines and replaced them by the following:

```
from torch.nn import BatchNorm2d
def InPlaceABNSync(input):
    return BatchNorm2d(input)
```
Thus supposedly using the standard torch BatchNorm2d instead of the InPlaceABNSync version, assuming InPlaceABNSync is just a BatchNorm2d function with no activation layer.

The code run without error, but if we take the demo example (`demo/demo.jpg`) using the lip dataset using the latest checkpoint (exp-schp-201908261155-lip.pth)t, it is far from what is expected:

![image](https://github.com/GoGoDuck912/Self-Correction-Human-Parsing/assets/7422936/548b4347-5349-4e4b-9764-fa1d065a8d54)

Instead of:

![image](https://github.com/GoGoDuck912/Self-Correction-Human-Parsing/assets/7422936/6cf03c90-c83a-4383-8e5b-d3207c7020b3)

The source image is indeed exactly the same, there is thus a difference somewhere in the code.

Can somebody help / show how to make the code work properly without the use of InPlaceABNSync?

Thank you,

**Comment by madanmanit** [comment]
Did you find a solution? I too am using a windows 11 env

**Comment by maxlegrec1** [comment]
You can define a custom module that combines `BatchNorm2d` with `LeakyReLU` like this:
```
from torch.nn import BatchNorm2d
from torch.nn import functional as F


def InPlaceABNSync(input):
    return LeakyReLUBatchNorm2d(input)


class LeakyReLUBatchNorm2d(nn.BatchNorm2d):
    def __init__(self, num_features, negative_slope=0.01, **kwargs):
        super(LeakyReLUBatchNorm2d, self).__init__(num_features, **kwargs)
        self.negative_slope = negative_slope

    def forward(self, input):
        # Apply BatchNorm2d as usual
        normalized = super().forward(input)
        # Apply LeakyReLU
        return F.leaky_relu(normalized, negative_slope=self.negative_slope)
```
